ballerina/tests/mock_services.bal

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+
+listener http:Listener mockListener = new (9090);
+
+service on mockListener {
+
+    # Adds an email to the allowlist.
+    #
+    # + caller - Represents the client making the request.
+    # + req - The HTTP request received by the service.
+    # + return - Returns an error if the operation fails.
+    resource function post allowlists/add(http:Caller caller, http:Request req) returns error? {
+        log:printInfo("Received a request to add to allowlist");
+        InlineResponse200 response = {
+            email: "test@example.com",
+            added: true
+        };
+        check caller->respond(response);
+    }
```

Comment:
* Instead of returning the `nil` and using the `caller` to respond, use the exact return type. 
* We don't need logs in the mock server.

Check similar places.
```suggestion
    resource function post allowlists/add() returns InlineResponse200|error {
        return {
            email: "test@example.com",
            added: true
        };
    }
```